src/httpclient.c

```diff
@@ -251,11 +251,15 @@ static void start_request(h2o_httpclient_ctx_t *ctx)
         return;
     }
 
+    if (req.connect_to)
+        req.connect_to->scheme = url_parsed->scheme;
+
     /* initiate the request */
     if (connpool == NULL) {
         connpool = h2o_mem_alloc(sizeof(*connpool));
         h2o_socketpool_t *sockpool = h2o_mem_alloc(sizeof(*sockpool));
-        h2o_socketpool_target_t *target = h2o_socketpool_create_target(url_parsed, NULL);
+        h2o_socketpool_target_t *target = h2o_socketpool_create_target(
+                req.connect_to ? req.connect_to : url_parsed, NULL);
```

Comment:
I think we have ambiguity regarding what `-x` option specifies. Does it specify the remote address at the transport layer, or does it _also_ specify the remote identity at the TLS layer (i.e. server name indicated by the certificate)?

I believe the intention shared among the two PRs have been the former, but this code works the latter way, because socket pool creates bidirectional streams to the designated target (regardless of that being directly on top of cleartext TCP or on top of TLS).

I do not this problem is a blocker for this PR, because h2o-httpclient is nothing more a test client and therefore we can always use the `-k` option.

However, I think it would be helpful to have that limitation be noted in the help message.

Reply:
@kazuho Thank you for pointing out the distinction.  You are correct in that the intention is to specify the transport layer address, but I did not pay attention to the TLS side effect.  Is there a simple way that you recommend to achieve this instead of using `h2o_socketpool_create_target()`?

Reply:
I think the only way of addressing the problem would to add a property to `h2o_socket_target_t`, so that server address (to which the TCP connection should be established) and server identity (to be verified by TLS) can be different.

That would be the ideal fix.

Though I'm not sure if we need to do that now due to the reasons stated above. Even if we decide to, to take small steps, I am fine with merging this PR knowing that such limitation exists, then addressing the problem in a separate PR.

Reply:
Thanks @kazuho.  I have added the comment in [2c7f76f4](https://github.com/h2o/h2o/pull/2691/commits/2c7f76f45e581ddc209ce216e368446c187b0032), and filed [h2o/h2o/issues/2698](https://github.com/h2o/h2o/issues/2698) to track the the additional enhancement.